lib/net_suite/client.rb

```diff
@@ -44,6 +48,19 @@ def #{method}(url = nil, body = nil, headers = nil, &block)
 
     attr_accessor :retry_count
 
+    def trace_call(method, url, request_payload, *args, &block)
+      return yield unless defined?(::Datadog::Tracing.trace)
+
+      started = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+
+      ::Datadog::Tracing.trace("netsuite #{method}",
+        resource: "#{method} #{url}",
+        span_type: 'http',
+        service: 'netsuite',
+        tags: { method: method, url: url, request_payload: request_payload },
```

Comment:
@ericgross, following up on the idea of making sync/async a tag, how does this look?

```suggestion
        tags: { method: method, url: url, request_payload: request_payload, async: defined?(Sidekiq.server?) && Sidekiq.server? },
```

Reply:
Could we refactor the value of `async` into a method to keep this line simpler?

Reply:
sure, added in https://github.com/careofvitamins/net_suite/pull/4/commits/0f73ddd818cb100970ed71f549ed6117cb3d7df2